src/shims/tsd.h

```diff
@@ -228,6 +296,8 @@ _dispatch_thread_setspecific_packed_pair(pthread_key_t k1, pthread_key_t k2,
 #if DISPATCH_USE_DIRECT_TSD
 #define _dispatch_thread_port() ((mach_port_t)(uintptr_t)\
 		_dispatch_thread_getspecific(_PTHREAD_TSD_SLOT_MACH_THREAD_SELF))
+#elif DISPATCH_USE_THREAD_LOCAL_STORAGE
+#define _dispatch_thread_port() ((mach_port_t)(_dispatch_get_tsd_base()->tid))
```

Comment:
Technically definining it for Windows is not the right way, but let's leave that part alone. it should also define `_dispatch_tid_self()` eventually if someone decides to re-port dispatch in the future, the same way I'm asking you to do for linux.